drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -88,8 +88,17 @@
 % gradient computations.
 kinsol.q = q;
 kinsol.v = v;
+kinsol.has_gradients = options.compute_gradients;
 
-if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
+if options.use_mex && model.mex_model_ptr~=0 && ((isnumeric(q) && isnumeric(v)) || (isa(q, 'TaylorVar') && getOrder(q) < 2)) % relying on short circuiting here
+  if ~isnumeric(q)
+    if isa(q, 'TaylorVar')
+      if options.compute_gradients
+        error('can only compute first order gradients using mex')
```

Comment:
i don't like this error message.  i think you're trying to avoid someone asking to compute the gradients of a taylorvar input (as opposed to using TaylorVar to compute the gradients of the nominal input).  but it took me a few minutes to parse the logic and figure out what you meant.